modules/programs/zsh.nix

```diff
@@ -74,13 +129,48 @@ in
       initExtra = mkOption {
         default = "";
         type = types.lines;
-        description = "Extra commands that should be added to .zshrc.";
+        description = "Extra commands that should be added to <filename>.zshrc</filename>.";
+      };
+
+      plugins = mkOption {
+        type = types.listOf pluginModule;
+        default = [];
+        example = literalExample ''
+          [
+            {
+              # will source zsh-autosuggestions.plugin.zsh
+              name = "zsh-autosuggestions";
+              src = pkgs.fetchFromGitHub {
+                owner = "zsh-users";
+                repo = "zsh-autosuggestions";
+                rev = "v0.4.0";
+                sha256 = "0z6i9wjjklb4lvr7zjhbphibsyx51psv50gm07mbb0kj9058j6kc";
+              };
+            }
+            {
+              file = "init.sh";
```

Comment:
Since `plugin.name` is now required a parameter, this example is not working anymore. Should probably add something like `name = "enhancd"`.